download/download.go

```diff
@@ -0,0 +1,181 @@
+package download
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/spf13/afero"
+)
+
+var afs = afero.Afero{Fs: afero.NewOsFs()}
+
+func init() {
+	log.Namespace = "dp-mongodb-in-memory"
+}
+
+// GetMongoDB returns the path to the mongod binary for the given config
+// It will download one if not found in the cache
```

Comment:
unsure why this function needs to return the cache path, as its in the config ? Also should we checksum again here ?

Reply:
Good point! There is really no need to return the path and I will change this to only return error.

Re the checksum, this function will call the `downloadMongoDB` which will verify the checksum if it needs to download the mongo binary. If it is in the cache, it has already been verified and can be trusted